pyroomacoustics/room.py

```diff
@@ -2189,15 +2212,230 @@ def extrude(self, height):
+        snr: float, optional
+            The target signal-to-noise ratio (SNR) in decibels at the reference microphone.
+            When this option is used the argument
+            :py:attr:`pyroomacoustics.room.Room.sigma2_awgn` is ignored. The variance of
+            every source at the reference microphone is normalized to one and
+            the variance of the noise \\(\\sigma_n^2\\) is chosen
+
+            .. math::
+
+                \mathsf{SNR} = 10 \log_{10} \frac{ K }{ \sigma_n^2 }
+
+            The value of :py:attr:`pyroomacoustics.room.Room.sigma2_awgn` is also set
+            to \\(\\sigma_n^2\\) automatically
+
+        callback_mix: func, optional
+            A function that will perform the mix, it takes as first argument
+            an array of shape ``(n_sources, n_mics, n_samples)`` that contains
+            the source signals convolved with the room impulse response prior
+            to mixture at the microphone. It should return an array of shape
+            ``(n_mics, n_samples)`` containing the mixed microphone signals.
+            If such a function is provided, the ``snr`` option is ignored
+            and :py:attr:`pyroomacoustics.room.Room.sigma2_awgn` is set to ``None``.
+        callback_mix_kwargs: dict, optional
+            A dictionary that contains optional arguments for ``callback_mix``
+            function
+        return_premix: bool, optional
+            If set to ``True``, the function will return an array of shape
+            ``(n_sources, n_mics, n_samples)`` containing the microphone
+            signals with individual sources, prior to mixing
+
+        Returns
+        -------
+        Nothing or an array of shape ``(n_sources, n_mics, n_samples)``
+            Depends on the value of ``return_premix`` option
         """
-        Computes the volume of a room
-        :param room: the room object
-        :return: the volume in cubic unit
-        """
+        # Throw an error if we are missing some hardware in the room
+        if len(self.sources) == 0:
+            raise ValueError("There are no sound sources in the room.")
+        if self.mic_array is None:
+            raise ValueError("There is no microphone in the room.")
 
-        return np.prod(self.shoebox_dim)
+        # number of mics and sources
+        M = self.mic_array.M
+        S = len(self.sources)
+
+        # compute all delays from M microphones to S sources, and 
+        # maximum signal length
+
+        max_sig_len = 0 
+
+        # TODO(FD) make this faster using broadcasting
+        delays = np.zeros((M, S), dtype=np.int)
+        for m in range(M):
+            for s in range(S):
+
+                source_delay = self.sources[s].delay * self.fs
+                mic_delay = self.get_delay(s, m)  * self.fs
+
+                #TODO(FD) use floor or round here? 
+                delay = int(source_delay + mic_delay)
+                delays[m, s] = delay
```

Comment:
This is a little detail, but I saw that sometimes the integer delay is calculated using the floor operation, instead of rounding. What would you like to use here?